tsdb/querier.go

```diff
@@ -364,7 +350,9 @@ func PostingsForMatchers(ix IndexReader, ms ...labels.Matcher) (index.Postings,
 				}
 				its = append(its, it)
 			}
-		} else { // l=""
+		} else { // l!="foo" or l=""
```

Comment:
As we are using `inversePostingsForMatcher` for `l!="foo"` here (instead of `postingsForMatcher` above), this could make queries including `l!="foo"` to perform worse than before.

But looking at the refactoring, it seems that the code always fell back here for `l!="foo"`.

Can you add another `} else if isNot {...}` case here which includes the above code that was removed (which includes `postingsForMatcher`)?

Reply:
@codesome Make sense, updated.